### PR TITLE
indexer reader: use raw_checkpoints instead for JSON RPC

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

title, the related issue is https://linear.app/mysten-labs/issue/DP-45/instead-of-having-tx-digests-we-should-just-put-raw-checkpointcontents

## Test plan 

ci

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Comment by lxfind** [comment]
We already added the `raw_checkpoints` table though. Wouldn't that be sufficient?

**Comment by gegaowp** [comment]
@lxfind maybe? honestly I prob missed contexts about the how we plan to use `raw_checkpoints` etc. and under the impression that it will eventually be in a separate kv store given PG's capacity limit and mainly serves history queries, please correct me if I am off.
I overall like the idea that pruned PG alone can still serve queries about recent data, so that operators do not have to worry about kv store within the data retention, not super opinionated on this though.

File: crates/sui-indexer/migrations/pg/2024-09-26-143533_tx_digests_to_checkpoint_contents/up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE checkpoints DROP COLUMN tx_digests;
```
Comment (lxfind) [inline crates/sui-indexer/migrations/pg/2024-09-26-143533_tx_digests_to_checkpoint_contents/up.sql:1]: This is not a safe breaking change. Please follow the schema change process.

File: crates/sui-indexer/migrations/pg/2024-09-26-143533_tx_digests_to_checkpoint_contents/up.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+ALTER TABLE checkpoints ADD COLUMN checkpoint_contents BYTEA;
```
Comment (bmwill) [inline crates/sui-indexer/migrations/pg/2024-09-26-143533_tx_digests_to_checkpoint_contents/up.sql:2]: This data already lives in the new checkpoints table so we can likely just follow through dropping the tx_digests column and omit adding this one

**Comment by gegaowp** [comment]
@lxfind @bmwill okay I read around and agreed that we can use raw_checkpoints instead, I re-purposed this pr to change indexer_reader to use `raw_checkpoints` instead for JSON RPC, the only usage of `checkpoints` is lookup from checkpoint digest to checkpoint seq. In other words, if we have a separate index table for this lookup, at least for JSON RPC, we should be able to deprecate `checkpoints` table as a whole.

File: crates/sui-indexer/src/types.rs [hunk 3]
```diff
@@ -44,7 +44,6 @@ pub struct IndexedCheckpoint {
     pub non_refundable_storage_fee: u64,
     pub checkpoint_commitments: Vec<CheckpointCommitment>,
     pub validator_signature: AggregateAuthoritySignature,
-    pub successful_tx_num: usize,
```
Comment (gegaowp) [inline crates/sui-indexer/src/types.rs:47]: this was for TPS computation, its no longer stored in DB or used.

**Comment by lxfind** [comment]
The table is still empty btw, I have not yet backfilled it.

**Comment by gegaowp** [comment]
@lxfind noted, will hold until the backfill is complete.

**Comment by lxfind** [comment]
btw I don't actually have a strong opinion on whether we want to duplicate the raw data in the indexed checkpoint table. I see merits in both ways. I think once we have the data, we could indeed consider adding it to the indexed checkpoint table, and if we find it not useful later we could always remote it.

**Pull request closed without merging**